server/coordinator/procedure/create_table.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package procedure
+
+import (
+	"context"
+	"sync"
+
+	"github.com/CeresDB/ceresdbproto/pkg/metaservicepb"
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/CeresDB/ceresmeta/server/coordinator/eventdispatch"
+	"github.com/looplab/fsm"
+	"github.com/pkg/errors"
+)
+
+const (
+	eventCreateTablePrepare = "EventCreateTablePrepare"
+	eventCreateTableFailed  = "EventCreateTableFailed"
+	eventCreateTableSuccess = "EventCreateTableSuccess"
+
+	stateCreateTableBegin   = "StateCreateTableBegin"
+	stateCreateTableWaiting = "StateCreateTableWaiting"
+	stateCreateTableFinish  = "StateCreateTableFinish"
+	stateCreateTableFailed  = "StateCreateTableFailed"
+)
+
+var (
+	createTableEvents = fsm.Events{
+		{Name: eventCreateTablePrepare, Src: []string{stateCreateTableBegin}, Dst: stateCreateTableWaiting},
+		{Name: eventCreateTableSuccess, Src: []string{stateCreateTableWaiting}, Dst: stateCreateTableFinish},
+		{Name: eventCreateTableFailed, Src: []string{stateCreateTableWaiting}, Dst: stateCreateTableFailed},
+	}
+	createTableCallbacks = fsm.Callbacks{
+		eventCreateTablePrepare: createTablePrepareCallback,
+		eventCreateTableFailed:  createTableFailedCallback,
+		eventCreateTableSuccess: createTableSuccessCallback,
+	}
+)
+
+func createTablePrepareCallback(event *fsm.Event) {
+	request := event.Args[0].(*CreateTableCallbackRequest)
```

Comment:
Shall check the length of the Args to avoid out-of-bound panic?

Reply:
The test already cover this, no need to check.